File: 0101-symmetric-tree/0101-symmetric-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {

    var checkSymmetric = function(left, right) {
        if(!left && !right) return true;
        if(!left || !right) return false;

        return checkSymmetric(left.left, right.right) && checkSymmetric(left.right, right.left) && left.val === right.val;
    }

    return checkSymmetric(root.right, root.left);

};

File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        return self.checkSymmetric(root.left, root.right)
    

    def checkSymmetric(self, left: Optional[TreeNode], right: Optional[TreeNode]) -> bool:
        if left == None and right == None:
            return True
        
        if left == None or right == None:
            return False

        return (self.checkSymmetric(left.left, right.right) and self.checkSymmetric(left.right, right.left) and left.val == right.val)